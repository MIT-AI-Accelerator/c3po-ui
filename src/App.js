import logo from './logo.svg';
import './App.css';
import React, { useState, useEffect } from 'react';
import OutlinedInput from '@mui/material/OutlinedInput';
import InputLabel from '@mui/material/InputLabel';
import InputAdornment from '@mui/material/InputAdornment';
import FormControl from '@mui/material/FormControl';
import { useCallback } from "react";
import Particles from "react-tsparticles";
import { loadFull } from "tsparticles";

import { submitPrompt } from './api/sentiments';

function App() {
  const [answer, setAnswer] = useState(null);
  const [prompt, setPrompt] = useState("");

  useEffect(() => {

    async function getAnswer(prompt) {
      const newAnswerObj = await submitPrompt(prompt);
      newAnswerObj.status === 200 && setAnswer(newAnswerObj.data.answer);
    }

    if (prompt) {
      setAnswer("");
      getAnswer(prompt);
    }

  }, [prompt]);


  const handlePromptEnter = (event) => {
    if (event.key === "Enter") {
      setPrompt(event.target.value);
    }
  };


  const particlesInit = useCallback(async (engine) => {
    console.log(engine);
    // you can initiate the tsParticles instance (engine) here, adding custom shapes or presets
    // this loads the tsparticles package bundle, it's the easiest method for getting everything ready
    // starting from v2 you can add only the features you need reducing the bundle size
    await loadFull(engine);
  }, []);

  const particlesLoaded = useCallback(async (container) => {
    await console.log(container);
  }, []);

  return (
    <div className="App">
      {/* <Particles id="tsparticles" url="http://foo.bar/particles.json" init={particlesInit} loaded={particlesLoaded} /> */}
      <header className="App-header">
        <FormControl fullWidth sx={{ m: 1 }}>
          <InputLabel htmlFor="outlined-adornment-amount">C3PO Prompt</InputLabel>
          <OutlinedInput
            id="outlined-adornment-amount"
            onKeyUp={handlePromptEnter}
            startAdornment={<InputAdornment position="start">>>>></InputAdornment>}
            label="C3PO Prompt"
          />
        </FormControl>
        {answer === null && <p>
          <code>Type your C3PO prompt and hit enter</code>
        </p>}
        {answer === "" && <p>
          <code>Loading...</code>
        </p>}
        {answer && <p>
          <code>{answer}</code>
        </p>}
      </header>
    </div>
  );
}

export default App;
