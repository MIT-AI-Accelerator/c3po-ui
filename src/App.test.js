import { render, screen, within } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('renders App component', () => {
    render(<App />);
    screen.debug();
  });

  it('Contains an input with a C3PO Prompt label', () => {
    render(<App />);
    const labelElement = within(document.getElementsByTagName("fieldset")[0]).getByText(/C3PO/);
    const inputElement = screen.getByRole("textbox");

    expect(labelElement).toBeInTheDocument();
    expect(inputElement).toBeInTheDocument();

  });

  it('Matches the last snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

});