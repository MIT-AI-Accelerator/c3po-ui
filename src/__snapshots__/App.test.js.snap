// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Matches the last snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="outlined-adornment-amount"
        >
          C3PO Prompt
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
            >
              &gt;&gt;&gt;&gt;
            </p>
          </div>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-adornment-amount"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                C3PO Prompt
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p>
        <code>
          Type your C3PO prompt and hit enter
        </code>
      </p>
    </header>
  </div>
</DocumentFragment>
`;
